src/Domain/Services/ExtendRestApi.php

```diff
@@ -118,6 +126,66 @@ public function register_endpoint_data( $args ) {
 		return true;
 	}
 
+	/**
+	 * Add callback functions that can be executed by the cart/extensions endpoint.
+	 *
+	 * @param array $args {
+	 *     An array of elements that make up the callback configuration.
+	 *
+	 *     @type string   $endpoint The endpoint to extend.
+	 *     @type string   $namespace Plugin namespace.
+	 *     @type callable $callback_function The function/callable to execute.
+	 * }
+	 *
+	 * @throws Exception On failure to register.
+	 * @returns boolean True on success.
+	 */
+	public function register_update_callback( $args ) {
+		if ( ! is_string( $args['namespace'] ) ) {
+			$this->throw_exception( 'You must provide a plugin namespace when extending a Store REST endpoint.' );
+		}
+
+		if ( ! is_string( $args['endpoint'] ) || ! in_array( $args['endpoint'], $this->endpoints, true ) ) {
+			$this->throw_exception(
+				sprintf( 'You must provide a valid Store REST endpoint to extend, valid endpoints are: %1$s. You provided %2$s.', implode( ', ', $this->endpoints ), $args['endpoint'] )
+			);
+		}
```

Comment:
I don't get the point of this option, I'm going to assume the only way people can register update callbacks is going to be in `cart/extensions`, so why add it to the options if it's going to be static?

Reply:
Correct, I've removed it, I don't foresee any other routes they could need to update the cart on for now.